docs/key_concepts.md

```diff
@@ -0,0 +1,55 @@
+## Key Concepts
+
+`whisk` is a ML project framework that make makes collaboration, reproducibility, and deployment "just work". Let's go through a few concepts that are important to understanding how `whisk` works.
```

Comment:
We should be more consistent in our usage of code styling (the ` or grave accent). I'm guilty of this too.

I think we should only wrap "whisk" in grave accents when we're referring to the actual command (ie `whisk create`), or code (ie `import whisk`). If we're referring to the project as a whole, just use whisk. It was distracting to me reading this as `whisk` really stood out.

Reply:
Ah yeah great call, I'll clean that up so that only code has `

While we're on this topic - if "whisk" is used at the beginning of a sentence, Im planning on _not_ capitalizing it. Do you agree with that?

Reply:
I'm good with all lowercase all the time.